daft/iceberg/iceberg_write.py

```diff
@@ -0,0 +1,180 @@
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+
+from daft import Series
+from daft.table import MicroPartition
+
+if TYPE_CHECKING:
+    import pyarrow as pa
+    from pyiceberg.io.pyarrow import _TablePartition
+    from pyiceberg.partitioning import PartitionSpec as IcebergPartitionSpec
+    from pyiceberg.schema import Schema as IcebergSchema
+    from pyiceberg.typedef import Record
+
+
+def _coerce_pyarrow_table_to_schema(pa_table: "pa.Table", schema: "IcebergSchema") -> "pa.Table":
```

Comment:
Why is this necessary to build in our code? I'm surprised that PyIceberg doesn't do this themselves since this is all pyiceberg/pyarrow code.

Reply:
I have not verified this but I couldn't find anything in the pyiceberg code for doing this coercion, I believe it simply fails if the schema differs. We had this function in our codebase before, I am just moving it here since it will only be used here.